File: 1089.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main{

  public static void main (String[] args) throws IOException{
    InputStreamReader go = new InputStreamReader(System.in);
    BufferedReader reader = new BufferedReader(go);

    int n = Integer.parseInt(reader.readLine());

    while(n > 0){

      String[] initial_data = reader.readLine().split(" ");
      int[] data = new int[n];

      for(int i = 0; i < n; i++){
      	data[i] = Integer.parseInt(initial_data[i]);
      }

      boolean down = false;
      boolean last = false;

      if(data[n-1] > data[0]){
      		last = true;
      }

      int peeks = 0;

      for(int j= 0; j < n-1; j++){
      	
      	if(data[j] > data[j+1]){
      		down = true;
      	} else {
      		down = false;
      	}

      	if(last != down){
      		peeks++;
      	}

      	last = down;
      }

      if(data[n-1] > data[0]){
      	down = true;
      } else {
      	down = false;
      }

      if(last!= down){
      	 peeks++;
      }

       System.out.println(peeks);
       n = Integer.parseInt(reader.readLine());

    }
  }
}

File: 1089.py
n = int(input())
while(n > 0):

	inp = input().split()

	for i in range(n):
		d={}
		c = 0

		for j in inp:
			d[c] = int(j)
			c= c+1

	p = 0


	if(d[c-1] >d[0]):
		 	last= True # I am going down 
			
	else:
			last = False	#I am going up 
	

	down= False

	for r in range(0, len(d)-1 ):

		if(d[r] > d[r + 1]):
			down = True
		else:
			down = False
		

		if(last != down):
			p = p +1

		last = down

	print(p+1)
	n = int(input())
		

		 	
